### PR TITLE
Ssp driver new

### PR DESCRIPTION
**Pull request by diarmadhi**

### CODE REVIEW

File: firmware/library/L0_LowLevel/LPC40xx.h [hunk 1]
```diff
@@ -1,13 +1,13 @@
 /****************************************************************************************************//**
-* $Id$		LPC407x_8x.h			2012-04-25
+* $Id$    LPC407x_8x.h      2012-04-25
```
[the remark this answers is not in the document]
Comment (diarmadhi) [reply firmware/library/L0_LowLevel/LPC40xx.h]: Reverted back to master

File: firmware/library/L1_Drivers/ssp.hpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
+
```
Comment (kammce) [inline firmware/library/L1_Drivers/ssp.hpp]: Missing include guard. #pragma once.

File: firmware/library/L1_Drivers/ssp.hpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
+
+
+#include "LPC17xx.h"
+
+class SSP
+{
+private: 
+	/*SSP register lookup table*/
+	LPC_SSP_TypeDef* SSP[3] = {LPC_SSP0, LPC_SSP1, LPC_SSP2};
```
Comment (kammce) [inline firmware/library/L1_Drivers/ssp.hpp]: See PinConfigure. You want to make this static and initialize in a cpp file for the purpose of dependency injection.

File: firmware/library/L1_Drivers/ssp.hpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+#include "LPC17xx.h"
+
+class SSP
+{
+private: 
+	/*SSP register lookup table*/
+	LPC_SSP_TypeDef* SSP[3] = {LPC_SSP0, LPC_SSP1, LPC_SSP2};
+
+	/*SSP pins*/
+
```
Comment (kammce) [inline firmware/library/L1_Drivers/ssp.hpp]: Nit: too many newlines. Google standard is to only have 1 newline.

**Comment by kammce** [comment]
Closing due to duplicate PR #115

**Pull request closed without merging**